### ISSUE TITLE
No way to download CSS automatically

### ISSUE THREAD

**Issue by milkywayfarer** [roflmuffin/CounterStrikeSharp#506]
Stuck with a problem that requires me to add a build tag along with the CSS version to my environment variables to pull the release to my server using a bash script. As it seems, GitHub Actions adds a tag to every release asset name, which makes "hard-wiring" a download link a chore (for example, MatchZy releases a zip asset named `MatchZy-${MATCHZY_VERSION}.zip`, whereas I need to do something along the lines of `counterstrikesharp-${CSS_VERSION}-build-${CSS_BUILDTAG}.zip` and manually change the tag every time a new release comes).

TLDR it would be nice to have a static link to a latest release (or include only a version in the asset name).

Thanks!

**Comment by RealGoodAidan** [comment]
This probably isn't what you're looking for, but I'm downloading the the latest MMSource / CS# releases with this brash script (added to my overridden pre.sh script):

```bash
#!/bin/bash

mkdir -p "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp/plugins/CounterShop

latest_mm_source_url="https://mms.alliedmods.net/mmsdrop/2.0/$(curl -L https://mms.alliedmods.net/mmsdrop/2.0/mmsource-latest-linux | xargs)"
latest_cssharp_url=$(curl -s https://api.github.com/repos/roflmuffin/CounterStrikeSharp/releases/latest | grep '"browser_download_url":' | grep '.*\-with\-runtime.*\-linux' | grep -vE '(\.pem|\.sig)' | grep -o 'https://[^"]*')

curl $latest_mm_source_url | tar xvzf - -C "${STEAMAPPDIR}"/game/csgo
curl -L -o "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp.zip $latest_cssharp_url
unzip -o "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp.zip -d "${STEAMAPPDIR}"/game/csgo/
rm "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp.zip

GAME_INFO_FILE="${STEAMAPPDIR}/game/csgo/gameinfo.gi"
LINE_TO_ADD="     Game csgo/addons/metamod"

# Check if the line already exists
if ! grep -qF "$LINE_TO_ADD" "$GAME_INFO_FILE"; then
    sed -i "/SearchPaths/{n; a\\
$LINE_TO_ADD
}" "$GAME_INFO_FILE"
fi
```

**Comment by milkywayfarer** [comment]
> This probably isn't what you're looking for, but I'm downloading the the latest MMSource / CS# releases with this brash script (added to my overridden pre.sh script):
> 
> ```shell
> #!/bin/bash
> 
> mkdir -p "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp/plugins/CounterShop
> 
> latest_mm_source_url="https://mms.alliedmods.net/mmsdrop/2.0/$(curl -L https://mms.alliedmods.net/mmsdrop/2.0/mmsource-latest-linux | xargs)"
> latest_cssharp_url=$(curl -s https://api.github.com/repos/roflmuffin/CounterStrikeSharp/releases/latest | grep '"browser_download_url":' | grep '.*\-with\-runtime.*\-linux' | grep -vE '(\.pem|\.sig)' | grep -o 'https://[^"]*')
> 
> curl $latest_mm_source_url | tar xvzf - -C "${STEAMAPPDIR}"/game/csgo
> curl -L -o "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp.zip $latest_cssharp_url
> unzip -o "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp.zip -d "${STEAMAPPDIR}"/game/csgo/
> rm "${STEAMAPPDIR}"/game/csgo/addons/counterstrikesharp.zip
> 
> GAME_INFO_FILE="${STEAMAPPDIR}/game/csgo/gameinfo.gi"
> LINE_TO_ADD="     Game csgo/addons/metamod"
> 
> # Check if the line already exists
> if ! grep -qF "$LINE_TO_ADD" "$GAME_INFO_FILE"; then
>     sed -i "/SearchPaths/{n; a\\
> $LINE_TO_ADD
> }" "$GAME_INFO_FILE"
> fi
> ```
> 
> edit: should have mentioned, the pre.sh script I'm referring to is from this project: https://github.com/joedwards32/CS2
> 
> But the important part for you is the curl statement to get the latest release version.

That's a nice script! It might be usable for CSSharp, MatchZy and CS2Fixes all together, I'll give it a try, thanks!
